basic_pitch/data/pipeline.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python
+# encoding: utf-8
+#
+# Copyright 2022 Spotify AB
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import os
+import uuid
+from typing import Dict, List, Tuple, Callable, Union
+
+import apache_beam as beam
+import tensorflow as tf
+from apache_beam.options.pipeline_options import PipelineOptions
+
+
+# Beacase beam.GroupIntoBatches isn't supported as of 2.29
+class Batch(beam.DoFn):
+    def __init__(self, batch_size):
+        self.batch_size = batch_size
+
+    def process(self, element):
+        for i in range(0, len(element), self.batch_size):
+            yield element[i : i + self.batch_size]
+
+
+class WriteBatchToTfRecord(beam.DoFn):
```

Comment:
Also can't remember why we have this instead of using [beam's builtin](https://beam.apache.org/releases/pydoc/2.27.0/apache_beam.io.tfrecordio.html#apache_beam.io.tfrecordio.WriteToTFRecord)

Reply:
custom naming? uuid instead of shard nameing? seems like it doesn't really matter but i'm just guessing here.